### PR TITLE
[Feature] [Transform-V2]Add Deduplicate Transform for internal engine

### PR DESCRIPTION
**Pull request by rewerma**
## Purpose of this pull request

Add Deduplicate transform for  the internal engine, for example configuration: 

```
source {
  Jdbc {
      url = "jdbc:mysql://localhost:3306/foo"
      driver = "com.mysql.cj.jdbc.Driver"
      connection_check_timeout_sec = 100
      user = "root"
      password = "******"
      # Must sort by the column for deduplicate
      query = "select id, name, c_time from fake order by name"
  }
}

transform {
  Deduplicate {
    duplicate_fields = ["name"]
  }
}

sink {
  Console {}
}
```

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by hailin0 — COMMENTED** [review]
add docs & e2e testcase

https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/transform-v2
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-transforms-v2-e2e/src/test/java/org/apache/seatunnel/e2e/transform

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/src/test/resources/deduplicate_transform.conf [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    }
+    # The source data must sorted by duplication fields
+    rows = [
+      { fields = [1, "Eric", 21], kind = "INSERT" }
+      { fields = [5, "Eric", 21], kind = "INSERT" }
+      { fields = [2, "Alex", 36], kind = "INSERT" }
+      { fields = [9, "Alex", 36], kind = "INSERT" }
+      { fields = [3, "Eddie", 36], kind = "INSERT" }
+    ]
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/src/test/resources/deduplicate_transform.conf:42]: add testcase
`
{ fields = [1, "A", 21], kind = "INSERT" }
{ fields = [2, "B", 36], kind = "INSERT" }
{ fields = [5, "A", 21], kind = "INSERT" }
`

File: docs/en/transform-v2/deduplicate.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+The data read from source is a table like this:
+
+| id | name     | age | 
+|----|----------|-----|
+| 1  | Joy Ding | 20  | 
+| 5  | Joy Ding | 20  | 
+| 2  | Kin Dom  | 14  | 
+| 9  | Kin Dom  | 14  |
+
+The source table data must sort by duplicate fields. For example, JDBC source:
```
Comment (hailin0) [inline docs/en/transform-v2/deduplicate.md:34]: The source table data must partition by duplicate fields.

Comment (hailin0) [reply docs/en/transform-v2/deduplicate.md:34]: When the data is partitioned, multiple transform objects will be created to read in parallel, and the data will be partitioned first and then sorted

Comment (rewerma) [reply docs/en/transform-v2/deduplicate.md:34]: It is unsupported for the partitioned data yet, unless the data partitioned by the duplicate fields first.

Comment (EricJoy2048) [reply docs/en/transform-v2/deduplicate.md:34]: > When the data is partitioned, multiple transform objects will be created to read in parallel, and the data will be partitioned first and then sorted

+1

Comment (hailin0) [reply docs/en/transform-v2/deduplicate.md:34]: too restrictive

**Pull request closed without merging**